core/src/test/java/org/bitcoinj/core/WalletTest.java

```diff
@@ -851,6 +852,418 @@ public void onWalletChanged(Wallet wallet) {
     }
 
     @Test
+    public void doubleSpendWeCreate() throws Exception {
+        // Test we keep pending double spends in IN_CONFLICT until one of them is included in a block
+        // and we handle reorgs and dependency chains properly.
+        // The following graph shows the txns we use in this test and how they are related 
+        // (Eg txA1 spends txARoot outputs, txC1 spends txA1 and txB1 outputs, etc).
+        // txARoot -> txA1 -+
+        //                  |--> txC1 -> txD1
+        // txBRoot -> txB1 -+
+        //
+        // txARoot -> txA2 -+
+        //                  |--> txC2 -> txD2
+        // txBRoot -> txB2 -+
+        //
+        // txARoot -> txA3
+        //
+        // txA1 is in conflict with txA2 and txA3. txB1 is in conflict with txB2. 
+        
+
+        CoinSelector originalCoinSelector = wallet.getCoinSelector();        
+        try {
+            wallet.setCoinSelector(AllowUnconfirmedCoinSelector.get());
+            final Address address = new ECKey().toAddress(params);
+
+            Transaction txARoot = sendMoneyToWallet(valueOf(10, 0), AbstractBlockChain.NewBlockType.BEST_CHAIN);
+            SendRequest a1Req = SendRequest.to(address, valueOf(1, 0));
+            a1Req.tx.addInput(txARoot.getOutput(0));
+            a1Req.shuffleOutputs = false;
+            wallet.completeTx(a1Req);
+            Transaction txA1 = a1Req.tx;
+            SendRequest a2Req = SendRequest.to(address, valueOf(2, 0));
+            a2Req.tx.addInput(txARoot.getOutput(0));
+            a2Req.shuffleOutputs = false;
+            wallet.completeTx(a2Req);
+            Transaction txA2 = a2Req.tx;
+            SendRequest a3Req = SendRequest.to(address, valueOf(3, 0));
+            a3Req.tx.addInput(txARoot.getOutput(0));
+            a3Req.shuffleOutputs = false;
+            wallet.completeTx(a3Req);
+            Transaction txA3 = a3Req.tx;
+            wallet.commitTx(txA1);
+            wallet.commitTx(txA2);
+            wallet.commitTx(txA3);
+
+            Transaction txBRoot = sendMoneyToWallet(valueOf(100, 0), AbstractBlockChain.NewBlockType.BEST_CHAIN);
+            SendRequest b1Req = SendRequest.to(address, valueOf(11, 0));
+            b1Req.tx.addInput(txBRoot.getOutput(0));
+            b1Req.shuffleOutputs = false;
+            wallet.completeTx(b1Req);
+            Transaction txB1 = b1Req.tx;
+            SendRequest b2Req = SendRequest.to(address, valueOf(22, 0));
+            b2Req.tx.addInput(txBRoot.getOutput(0));
+            b2Req.shuffleOutputs = false;
```

Comment:
Perhaps we should default shuffleOutputs to false for unit test params/testnet - might reduce a lot of the duplication here.